realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -293,6 +293,18 @@ public int hashCode() {
         }
     }
 
+    public boolean waitForChange() {
+        return nativeWaitForChange(nativePtr);
+    }
+
+    public void setWaitForChangeEnabled(boolean enabled) {
+        if (enabled) {
+            nativeEnableWaitForChange(nativePtr);
```

Comment:
Instead of making two native methods, you could just make one with a boolean like you do here - and a 1:1 call then. Seems arbitrary to make two different API's.